src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -420,42 +428,52 @@ case class WdlBinding(ast: Ast, source: WdlSource, importResolver: ImportResolve
           }
 
           /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
-          val memberAccesses = WdlBinding.findAstsWithTrail(inputKv._2.ast, "MemberAccess") filter {
-            case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
-          }
-          memberAccesses.keys map validateMemberAccess
+          WdlBinding.findTopLevelMemberAccesses(inputKv._2.ast).map(getCallFromMemberAccessAst).map {_.get}
         }
       }
     }
   }
 
   /* Partially evaluate MemberAccess ASTs to make sure they're not nonsense at compile time */
-  def validateMemberAccess(ast: Ast): Any = {
+  def getCallFromMemberAccessAst(ast: Ast): Try[Call] = {
+    def callFromName(name: String): Try[Call] = {
+      workflows.head.calls.find(_.name == name) match {
+        case Some(c:Call) => Success(c)
+        case _ => Failure(new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast)))
+      }
+    }
     val rhs = sourceString(ast.getAttribute("rhs"))
     val lhs = ast.getAttribute("lhs") match {
-      case a: Ast => validateMemberAccess(a)
-      case terminal: Terminal =>
-        val name = sourceString(terminal)
-        val bindings = importedBindings filter {b => b.namespace == Some(name)}
-        val taskList = tasks filter {task => task.name == name}
-        val all = bindings ++ taskList
-        if (all.size == 0) throw new SyntaxError(wdlSyntaxErrorFormatter.undefinedMemberAccess(ast))
-        all.head
+      /**
```

Comment:
My preference would be for the comment to go above the `val lhs` portion but I don't care a ton